openAPI/core-back-internal.yaml

```diff
@@ -73,6 +73,59 @@ paths:
                   $bodyObj
                 #end
 
+  /cri/callback:
+    post:
+      description: |
+        Called when a user comes back to core on the frontend's callback endpoint, after visiting a CRI. Triggers a 
+        lambda for validating the oauth callback, retrieving the access token and fetching the credential.
+      responses:
+        200:
+          description: "Returns a journey or error response"
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/journeyType"
+      x-amazon-apigateway-integration:
+        type: "aws"
+        httpMethod: "POST"
+        uri:
+          Fn::Sub: arn:aws:apigateway:${AWS::Region}:lambda:path/2015-03-31/functions/${ProcessCriCallbackFunction.Arn}:live/invocations
+        passthroughBehavior: "when_no_match"
+        requestTemplates:
+          application/json:
+            Fn::Sub: |
+              #set($inputRoot = $input.path('$'))
+              {
+                "input": "{
+                  \"ipvSessionId\": \"$input.params('ipv-session-id')\", 
+                  \"featureSet\": \"$input.params('feature-set')\", 
+                  \"ipAddress\": \"$input.params('ip-address')\" #foreach($key in $inputRoot.keySet()),
+                  \"$key\": \"$inputRoot.get($key)\"#end
```

Comment:
Have you managed to test this yet? I wonder if instead you want to just pass the request directly to the lambda, and parse the request there (similar to e.g. the `InitializeIpvSessionHandler`)?

Reply:
I've been trying to set test it this morning. Hmm, I hadn't considered that. I'll leave this unresolved because might look into it once it's working